cmd/nerdctl/run_windows.go

```diff
@@ -71,9 +80,52 @@ func setPlatformOptions(
 		opts = append(opts, oci.WithMemoryLimit(uint64(mem64)))
 	}
 
+	// TODO implement hyper-v isolation
+	isolation, err := cmd.Flags().GetString("isolation")
+	if err != nil {
+		return nil, err
+	}
+	switch isolation {
+	case "host":
+		hpAnnotations := map[string]string{}
+		hpAnnotations[hostProcessContainer] = "true"
+
+		// If user is set we will attempt to start container with that user (must be present on the host)
+		// Otherwise we will inherit permissions from the user that the containerd process is running as
+		user, err := cmd.Flags().GetString("user")
+		if err != nil {
+			return nil, err
+		}
+		if user == "" {
+			hpAnnotations[hostProcessInheritUser] = "true"
+		}
+
+		opts = append(opts, oci.WithAnnotations(hpAnnotations))
+	case "default":
+		// no op
+		// use containerd's default runtime options
+	case "process":
+		opts = append(opts, WithWindowsProcessIsolated())
+	default:
+		// no op
+		// use containerd's default
+	}
```

Comment:
I think we should combine default and process cases and get rid of the WithWindowsProcessIsolated option as the spec is only going to have that filled in if we set it anyways which would seem like a bug. 

```suggestion
	switch isolation {
	case "host":
		hpAnnotations := map[string]string{
		        hostProcessContainer: "true",
		}

		// If user is set we will attempt to start container with that user (must be present on the host)
		// Otherwise we will inherit permissions from the user that the containerd process is running as
		user, err := cmd.Flags().GetString("user")
		if err != nil {
			return nil, err
		}
		if user == "" {
			hpAnnotations[hostProcessInheritUser] = "true"
		}

		opts = append(opts, oci.WithAnnotations(hpAnnotations))
	case "default", "process":
		// no op
		// use containerd's default runtime options
	default:
		return nil, fmt.Errorf("unknown isolation value %q. valid values are 'host', 'process' or 'default'", isolation)
	}
```

Reply:
Also, on docker this flag was defined globally I believe and would error on Linux for an invalid value, should we do this parsing in run.go instead?

Reply:
Ok, I remember why I did it this way now.

If the `default_runtime` is set in the `config.toml` to be hyper-v don't we need to override the default runtime settings?  aka, blank out hyper-v part of the spec?  

Maybe I don't quite understand it properly...

Reply:
Not much we could do it seems as hyperv is set in the shim itself if that's the case https://github.com/microsoft/hcsshim/blob/b295b1a866b1263ddede4f00dddf7a129b96c114/cmd/containerd-shim-runhcs-v1/service_internal.go#L109-L116

Reply:
https://github.com/containerd/nerdctl/pull/795#discussion_r810165333

Reply:
Oh missed that, yep yep. Yea we can do best effort here as we can't control what happens there really.. Maybe could have docs on this that if your default runtime is runhcs-hypervisor that isolation defaults to hyperv like it would for docker desktop

Reply:
I did call this out in the readme section with `default_runtime` set in config.toml. Maybe it could be clearer?

So I should clear out the hyper-v spec and set the windows part? or should I nil both out?

```golang
func WithWindowsProcessIsolated() oci.SpecOpts {
	return func(_ context.Context, _ oci.Client, _ *containers.Container, s *specs.Spec) error {
		if s.Windows == nil {
			s.Windows = &specs.Windows{} <=== should Set this or set it to nil?
		}
		if s.Windows.HyperV != nil {
			s.Windows.HyperV = nil
		}
		return nil
	}
}
```

Reply:
Ahh, missed that 😬 But there's no point in clearing the spec I think, as if they have runhcs-hypervisor as the default runtime the shim will set the hyperv section back anyways. We're setting all of this up before the shims launched so it seems moot

Reply:
> But there's no point in clearing the spec I think, as if they have runhcs-hypervisor as the default runtime the shim will set the hyperv section back anyway

So this means if the default runtime is hyperv then you can't run process isolated?  I not sure we can handle this from here.  doing nothing when `process` and `hyperv` is the default runtime would be bad experience as well.

Reply:
I've added this back... Let me know what you think

Reply:
Wait now that I'm thinking about it, isn't the default_runtime_name a CRI field? That shouldn't matter or get read here right? @AkihiroSuda

You can change the runtime as a whole via --runtime (although process, hyperv and LCOW are all wrapped up in the same shim on Windows so it doesn't matter really). The shim options would have to change for this to matter, and it seems like the only runtime whose options are configured is runc https://github.com/containerd/nerdctl/blob/main/cmd/nerdctl/run_runtime.go#L34-L63. But regardless, nil'ing out hyperv is still not needed, as we're the ones controlling the spec here. If the shim ends up setting it for some reason ([like here](https://github.com/microsoft/hcsshim/blob/b295b1a866b1263ddede4f00dddf7a129b96c114/cmd/containerd-shim-runhcs-v1/service_internal.go#L109-L116)) it's irrelevant what we did here

Reply:
> If the shim ends up setting it for some reason ([like here](https://github.com/microsoft/hcsshim/blob/b295b1a866b1263ddede4f00dddf7a129b96c114/cmd/containerd-shim-runhcs-v1/service_internal.go#L109-L116)) it's irrelevant what we do here

So this means if the default runtime is `hyperv` and the user runs `nerdctl --isolation process` the user won't get a process isolated container?  Or is it only if sim options are set?